### PR TITLE
Fixes #21: Align action processor with client-side action dispatcher

### LINKED ISSUE #21: OperationActionHandler makes it hard to add custom operation action handlers
Currently, the `OperationActionHandler` handles operation actions in the following way:
```java
@Override
public Optional<Action> execute(Action action, GraphicalModelState modelState) {
	switch (action.getKind()) {
	case Action.Kind.CREATE_NODE_OPERATION:
	case Action.Kind.CREATE_CONNECTION_OPERATION:
	case Action.Kind.RECONNECT_CONNECTION_OPERATION:
	case Action.Kind.REROUTE_CONNECTION_OPERATION:
	case Action.Kind.DELETE_ELEMENT_OPERATION:
	case Action.Kind.CHANGE_BOUNDS_OPERATION:
	case Action.Kind.APPLY_LABEL_EDIT_OPERATION:
		return doHandle((AbstractOperationAction) action, modelState);
	default:
		return Optional.empty();
	}
}
```

That means, if users would like to add custom operation actions, they'll have to overwrite the `OperationActionHandler` and add their new action kind in the check above. This is unexpected and also unnecessary. We could do the following instead:

```java
@Override
public Optional<Action> execute(final Action action, final GraphicalModelState modelState)  {
  if (action instanceof AbstractOperationAction && operationHandlerProvider.isHandled((AbstractOperationAction)action)) {
      return doHandle((AbstractOperationAction)action, modelState);
  }
  return Optional.empty();
}
```

### PR DESCRIPTION
**Pull request by martin-fleck-at**
- Action processor should invoke ALL registered action handlers
- Each response by an action handler should be processed
- Introduce client action handler that sends actions to client

Note: This change does not adapt the operation handlers which execute operations and do not return any actions as of now.

Signed-off-by: Martin Fleck <mfleck@eclipsesource.com>

### CODE REVIEW

File: plugins/org.eclipse.glsp.server/src/org/eclipse/glsp/server/di/MultiBindingDefaults.java [hunk 1]
```diff
@@ -88,7 +97,22 @@ private MultiBindingDefaults() {}
       TriggerEdgeCreationAction.class,
       UpdateModelAction.class);
 
+   public static final List<Class<? extends Action>> DEFAULT_CLIENT_ACTIONS = Lists.newArrayList(
+      SetClipboardDataAction.class,
```
Comment (tortmayr) [inline plugins/org.eclipse.glsp.server/src/org/eclipse/glsp/server/di/MultiBindingDefaults.java:101]: It looks like this list is not complete. There are a couple of other client actions (ike. Center, FitToScreen, Select, Expand, ServerStatusAction etc.)

We should also double check the ActionHandlers e.g. the `SetEditModeActionHandler` implicitly registers the `SetEditModeAction` which has to be registered as client action as well.

File: plugins/org.eclipse.glsp.server/src/org/eclipse/glsp/server/action/DefaultActionProcessor.java [hunk 2]
```diff
@@ -15,23 +15,26 @@
...
 import org.eclipse.glsp.api.jsonrpc.GLSPClientProvider;
 import org.eclipse.glsp.api.model.GraphicalModelState;
 import org.eclipse.glsp.api.model.ModelStateProvider;
 import org.eclipse.glsp.api.registry.ActionHandlerRegistry;
+import org.eclipse.glsp.server.actionhandler.RequestEditValidationHandler;
 
 import com.google.inject.Inject;
 
 public class DefaultActionProcessor implements ActionProcessor {
+   private static Logger LOG = Logger.getLogger(RequestEditValidationHandler.class);
```
[the remark this answers is not in the document]
Comment (martin-fleck-at) [reply plugins/org.eclipse.glsp.server/src/org/eclipse/glsp/server/action/DefaultActionProcessor.java]: Fixed ;-)

File: plugins/org.eclipse.glsp.server/src/org/eclipse/glsp/server/di/MultiBindingDefaults.java (hunk 1 above)
Comment (martin-fleck-at) [reply plugins/org.eclipse.glsp.server/src/org/eclipse/glsp/server/di/MultiBindingDefaults.java:107]: Very good point! As discussed, I registered the necessary actions as client actions and also removed up some unused actions, cf also https://github.com/eclipse-glsp/glsp-client/pull/74.

I also added a logging example for the server-side action dispatching, cf. https://github.com/eclipse-glsp/glsp-examples/pull/60

Comment (martin-fleck-at) [reply plugins/org.eclipse.glsp.server/src/org/eclipse/glsp/server/di/MultiBindingDefaults.java:91]: Very good observation! I tested this and you are right, the DIActionRegistry already captures all handled actions through the action and operation handler bindings. As a result, the default action binding was removed altogether. I pushed a new commit to reflect that. Thank you!

**Pull request merged**